### ISSUE TITLE
Move "WebContents" to src/main/webapp

### ISSUE THREAD

**Issue by gunnarmorling** [mincong-h/gsoc-hsearch#36]
That's where it should be as per Maven conventions.

**Comment by mincong-h** [comment]
Our teacher never taught us about it 😢  I'll try to fix it.

**Comment by mincong-h** [comment]
In the current structure, webapp is located under :

    ./WebContent

In the future structure, webapp will be located under : 

    ./src/main/webapp

The first is recognizable as standard Eclipse project structure and the second is recognizable as standard Maven project structure.

**Comment by mincong-h** [comment]
Everything seems to be find, but when I run the project under WildFly 10.0. But the home page `./src/main/webapp/index.html` was not loaded correctly. The web page show the below error message _Forbidden_.

<img width="1030" alt="screen shot 2016-06-13 at 21 18 52" src="https://cloud.githubusercontent.com/assets/10179217/16020066/8c4ea0e6-31ac-11e6-9a8e-341cfd9e2622.png">

Here's the complete log.

```
21:14:03,128 INFO  [org.jboss.modules] (main) JBoss Modules version 1.5.1.Final
21:14:04,037 INFO  [org.jboss.msc] (main) JBoss MSC version 1.2.6.Final
21:14:04,114 INFO  [org.jboss.as] (MSC service thread 1-6) WFLYSRV0049: WildFly Full 10.0.0.Final (WildFly Core 2.0.10.Final) starting
21:14:05,404 INFO  [org.jboss.as.server.deployment.scanner] (DeploymentScanner-threads - 1) WFLYDS0004: Found jsr352.war in deployment directory. To trigger deployment create a file called jsr352.war.dodeploy
21:14:05,470 INFO  [org.jboss.as.server] (Controller Boot Thread) WFLYSRV0039: Creating http management service using socket-binding (management-http)
21:14:05,498 INFO  [org.xnio] (MSC service thread 1-5) XNIO version 3.3.4.Final
21:14:05,508 INFO  [org.xnio.nio] (MSC service thread 1-5) XNIO NIO Implementation Version 3.3.4.Final
21:14:05,569 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 33) WFLYJCA0004: Deploying JDBC-compliant driver class org.h2.Driver (version 1.3)
21:14:05,574 INFO  [org.wildfly.extension.io] (ServerService Thread Pool -- 37) WFLYIO001: Worker 'default' has auto-configured to 8 core threads with 64 task threads based on your 4 available processors
21:14:05,576 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 38) WFLYCLINF0001: Activating Infinispan subsystem.
21:14:05,608 INFO  [org.jboss.as.connector] (MSC service thread 1-1) WFLYJCA0009: Starting JCA Subsystem (WildFly/IronJacamar 1.3.2.Final)
21:14:05,614 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-4) WFLYJCA0018: Started Driver service with driver-name = h2
21:14:05,627 INFO  [org.jboss.as.connector.subsystems.datasources] (ServerService Thread Pool -- 33) WFLYJCA0005: Deploying non-JDBC-compliant driver class com.mysql.jdbc.Driver (version 5.1)
21:14:05,629 INFO  [org.jboss.as.connector.deployers.jdbc] (MSC service thread 1-3) WFLYJCA0018: Started Driver service with driver-name = mysql
21:14:05,653 INFO  [org.jboss.as.jsf] (ServerService Thread Pool -- 44) WFLYJSF0007: Activated the following JSF Implementations: [main]
21:14:05,654 INFO  [org.jboss.as.naming] (ServerService Thread Pool -- 46) WFLYNAM0001: Activating Naming Subsystem
21:14:05,813 INFO  [org.jboss.as.naming] (MSC service thread 1-3) WFLYNAM0003: Starting Naming Service
21:14:05,814 INFO  [org.jboss.as.mail.extension] (MSC service thread 1-4) WFLYMAIL0001: Bound mail session [java:jboss/mail/Default]
21:14:05,841 WARN  [org.jboss.as.txn] (ServerService Thread Pool -- 54) WFLYTX0013: Node identifier property is set to the default value. Please make sure it is unique.
21:14:05,882 INFO  [org.jboss.as.webservices] (ServerService Thread Pool -- 56) WFLYWS0002: Activating WebServices Extension
21:14:05,886 INFO  [org.jboss.remoting] (MSC service thread 1-4) JBoss Remoting version 4.0.18.Final
21:14:05,890 INFO  [org.jboss.as.security] (ServerService Thread Pool -- 53) WFLYSEC0002: Activating Security Subsystem
21:14:05,896 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-3) WFLYUT0003: Undertow 1.3.15.Final starting
21:14:05,896 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 55) WFLYUT0003: Undertow 1.3.15.Final starting
21:14:05,900 INFO  [org.jboss.as.security] (MSC service thread 1-1) WFLYSEC0001: Current PicketBox version=4.9.4.Final
21:14:06,095 INFO  [org.jboss.as.ejb3] (MSC service thread 1-4) WFLYEJB0482: Strict pool mdb-strict-max-pool is using a max instance size of 16 (per class), which is derived from the number of CPUs on this host.
21:14:06,095 INFO  [org.jboss.as.ejb3] (MSC service thread 1-5) WFLYEJB0481: Strict pool slsb-strict-max-pool is using a max instance size of 64 (per class), which is derived from thread worker pool sizing.
21:14:06,157 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 55) WFLYUT0014: Creating file handler for path '/Users/mincong/Documents/wildfly-10.0.0.Final/welcome-content' with options [directory-listing: 'false', follow-symlink: 'false', case-sensitive: 'true', safe-symlink-paths: '[]']
21:14:06,236 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-4) WFLYUT0012: Started server default-server.
21:14:06,239 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-5) WFLYUT0018: Host default-host starting
21:14:06,432 INFO  [org.wildfly.extension.undertow] (MSC service thread 1-2) WFLYUT0006: Undertow HTTP listener default listening on 127.0.0.1:8080
21:14:06,524 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-6) WFLYJCA0001: Bound data source [java:jboss/datasources/ExampleDS]
21:14:06,525 INFO  [org.jboss.as.connector.subsystems.datasources] (MSC service thread 1-3) WFLYJCA0001: Bound data source [java:/MySqlDS-addresses]
21:14:06,739 INFO  [org.jboss.as.server.deployment.scanner] (MSC service thread 1-4) WFLYDS0013: Started FileSystemDeploymentService for directory /Users/mincong/Documents/wildfly-10.0.0.Final/standalone/deployments
21:14:06,804 INFO  [org.jboss.as.server.deployment] (MSC service thread 1-3) WFLYSRV0027: Starting deployment of "jsr352.war" (runtime-name: "jsr352.war")
21:14:07,104 INFO  [org.infinispan.factories.GlobalComponentRegistry] (MSC service thread 1-5) ISPN000128: Infinispan version: Infinispan 'Mahou' 8.1.0.Final
21:14:07,105 INFO  [org.infinispan.factories.GlobalComponentRegistry] (MSC service thread 1-7) ISPN000128: Infinispan version: Infinispan 'Mahou' 8.1.0.Final
21:14:07,195 INFO  [org.jboss.ws.common.management] (MSC service thread 1-1) JBWS022052: Starting JBossWS 5.1.3.Final (Apache CXF 3.1.4) 
21:14:08,689 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-3) WFLYSRV0059: Class Path entry jaxb-api.jar in /Users/mincong/Documents/wildfly-10.0.0.Final/standalone/deployments/jsr352.war/WEB-INF/lib/com.sun.tools.xjc_2.2.0.jar  does not point to a valid jar for a Class-Path reference.
21:14:08,692 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-3) WFLYSRV0059: Class Path entry jaxb-impl.jar in /Users/mincong/Documents/wildfly-10.0.0.Final/standalone/deployments/jsr352.war/WEB-INF/lib/com.sun.tools.xjc_2.2.0.jar  does not point to a valid jar for a Class-Path reference.
21:14:08,693 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-3) WFLYSRV0059: Class Path entry jsr173_1.0_api.jar in /Users/mincong/Documents/wildfly-10.0.0.Final/standalone/deployments/jsr352.war/WEB-INF/lib/com.sun.tools.xjc_2.2.0.jar  does not point to a valid jar for a Class-Path reference.
21:14:08,699 WARN  [org.jboss.as.server.deployment] (MSC service thread 1-3) WFLYSRV0059: Class Path entry activation.jar in /Users/mincong/Documents/wildfly-10.0.0.Final/standalone/deployments/jsr352.war/WEB-INF/lib/com.sun.tools.xjc_2.2.0.jar  does not point to a valid jar for a Class-Path reference.
21:14:08,950 INFO  [org.jboss.as.jpa] (MSC service thread 1-6) WFLYJPA0002: Read persistence.xml for us-address
21:14:09,487 INFO  [org.jboss.as.jpa] (ServerService Thread Pool -- 58) WFLYJPA0010: Starting Persistence Unit (phase 1 of 2) Service 'jsr352.war#us-address'
21:14:09,516 INFO  [org.hibernate.jpa.internal.util.LogHelper] (ServerService Thread Pool -- 58) HHH000204: Processing PersistenceUnitInfo [
	name: us-address
	...]
21:14:09,590 INFO  [org.jboss.weld.deployer] (MSC service thread 1-8) WFLYWELD0003: Processing weld deployment jsr352.war
21:14:09,694 INFO  [org.hibernate.validator.internal.util.Version] (MSC service thread 1-8) HV000001: Hibernate Validator 5.2.3.Final
21:14:09,729 INFO  [org.hibernate.Version] (ServerService Thread Pool -- 58) HHH000412: Hibernate Core {5.0.7.Final}
21:14:09,731 INFO  [org.hibernate.cfg.Environment] (ServerService Thread Pool -- 58) HHH000206: hibernate.properties not found
21:14:09,733 INFO  [org.hibernate.cfg.Environment] (ServerService Thread Pool -- 58) HHH000021: Bytecode provider name : javassist
21:14:09,788 INFO  [org.hibernate.orm.deprecation] (ServerService Thread Pool -- 58) HHH90000001: Found usage of deprecated setting for specifying Scanner [hibernate.ejb.resource_scanner]; use [hibernate.archive.scanner] instead
21:14:09,857 INFO  [org.hibernate.annotations.common.Version] (ServerService Thread Pool -- 58) HCANN000001: Hibernate Commons Annotations {5.0.1.Final}
21:14:09,922 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-8) WFLYEJB0473: JNDI bindings for session bean named 'IndexSession' in deployment unit 'deployment "jsr352.war"' are as follows:

	java:global/jsr352/IndexSession!io.github.mincongh.session.IndexSession
	java:app/jsr352/IndexSession!io.github.mincongh.session.IndexSession
	java:module/IndexSession!io.github.mincongh.session.IndexSession
	java:global/jsr352/IndexSession
	java:app/jsr352/IndexSession
	java:module/IndexSession

21:14:09,922 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-8) WFLYEJB0473: JNDI bindings for session bean named 'BatchSession' in deployment unit 'deployment "jsr352.war"' are as follows:

	java:global/jsr352/BatchSession!io.github.mincongh.session.BatchSession
	java:app/jsr352/BatchSession!io.github.mincongh.session.BatchSession
	java:module/BatchSession!io.github.mincongh.session.BatchSession
	java:global/jsr352/BatchSession
	java:app/jsr352/BatchSession
	java:module/BatchSession

21:14:09,922 INFO  [org.jboss.as.ejb3.deployment] (MSC service thread 1-8) WFLYEJB0473: JNDI bindings for session bean named 'SearchSession' in deployment unit 'deployment "jsr352.war"' are as follows:

	java:global/jsr352/SearchSession!io.github.mincongh.session.SearchSession
	java:app/jsr352/SearchSession!io.github.mincongh.session.SearchSession
	java:module/SearchSession!io.github.mincongh.session.SearchSession
	java:global/jsr352/SearchSession
	java:app/jsr352/SearchSession
	java:module/SearchSession

21:14:10,254 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 58) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
21:14:10,256 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 58) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
21:14:10,263 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 58) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
21:14:10,264 INFO  [org.infinispan.configuration.cache.EvictionConfigurationBuilder] (ServerService Thread Pool -- 58) ISPN000152: Passivation configured without an eviction policy being selected. Only manually evicted entities will be passivated.
21:14:10,276 INFO  [org.jboss.weld.deployer] (MSC service thread 1-2) WFLYWELD0006: Starting Services for CDI deployment: jsr352.war
21:14:10,331 INFO  [org.jboss.weld.Version] (MSC service thread 1-2) WELD-000900: 2.3.2 (Final)
21:14:10,423 INFO  [org.jboss.weld.deployer] (MSC service thread 1-4) WFLYWELD0009: Starting weld service for deployment jsr352.war
21:14:10,702 INFO  [org.jboss.as.jpa] (ServerService Thread Pool -- 59) WFLYJPA0010: Starting Persistence Unit (phase 2 of 2) Service 'jsr352.war#us-address'
21:14:10,829 INFO  [org.jboss.as.clustering.infinispan] (ServerService Thread Pool -- 58) WFLYCLINF0002: Started client-mappings cache from ejb container
21:14:11,171 INFO  [org.hibernate.dialect.Dialect] (ServerService Thread Pool -- 59) HHH000400: Using dialect: org.hibernate.dialect.MySQLDialect
21:14:11,232 INFO  [org.hibernate.envers.boot.internal.EnversServiceImpl] (ServerService Thread Pool -- 59) Envers integration enabled? : true
21:14:11,792 INFO  [org.hibernate.search.engine.Version] (ServerService Thread Pool -- 59) HSEARCH000034: Hibernate Search 5.5.1.Final
21:14:11,968 INFO  [org.hibernate.hql.internal.QueryTranslatorFactoryInitiator] (ServerService Thread Pool -- 59) HHH000397: Using ASTQueryTranslatorFactory
21:14:12,115 WARN  [org.hibernate.search.engine.impl.ConfigContext] (ServerService Thread Pool -- 59) HSEARCH000075: Configuration setting hibernate.search.lucene_version was not specified: using LUCENE_CURRENT.
21:14:13,553 INFO  [org.wildfly.extension.undertow] (ServerService Thread Pool -- 62) WFLYUT0021: Registered web context: /jsr352
21:14:13,597 INFO  [org.jboss.as.server] (ServerService Thread Pool -- 34) WFLYSRV0010: Deployed "jsr352.war" (runtime-name : "jsr352.war")
21:14:13,804 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0060: Http management interface listening on http://127.0.0.1:9990/management
21:14:13,804 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0051: Admin console listening on http://127.0.0.1:9990
21:14:13,805 INFO  [org.jboss.as] (Controller Boot Thread) WFLYSRV0025: WildFly Full 10.0.0.Final (WildFly Core 2.0.10.Final) started in 11170ms - Started 464 of 732 services (383 services are lazy, passive or on-demand)
```

**Comment by mincong-h** [comment]
According to the [Maven • WAR Plugin][1], the project structure looks like this : 

```
 |-- pom.xml
 `-- src
     `-- main
         |-- java
         |   `-- com
         |       `-- example
         |           `-- projects
         |               `-- SampleAction.java
         |-- resources
         |   `-- images
         |       `-- sampleimage.jpg
         `-- webapp
             |-- WEB-INF
             |   `-- web.xml
             |-- index.jsp
             `-- jsp
                 `-- websource.jsp
```

I think the webapp `jsr352` does respect the structure / convention.

[1]: http://maven.apache.org/plugins/maven-war-plugin/usage.html

**Comment by mincong-h** [comment]
This below issue has been reported and fixed in #42 Failed to access home page after moving WebContent to src/main/webapp.
